File: src/main/java/com/snackgame/server/applegame/business/domain/game/BoardConverter.java
package com.snackgame.server.applegame.business.domain.game;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.PropertyAccessor;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

@Converter(autoApply = true)
public class BoardConverter implements AttributeConverter<Board, String> {

    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();

    static {
        OBJECT_MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
        OBJECT_MAPPER.setVisibility(PropertyAccessor.IS_GETTER, JsonAutoDetect.Visibility.NONE);
    }

    @Override
    public String convertToDatabaseColumn(Board board) {
        try {
            return OBJECT_MAPPER.writeValueAsString(board);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Board convertToEntityAttribute(String dbJson) {
        try {
            return OBJECT_MAPPER.readValue(dbJson, Board.class);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/com/snackgame/server/game/snackgame/domain/BoardConverter.kt
package com.snackgame.server.game.snackgame.domain

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.annotation.PropertyAccessor
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class BoardConverter : AttributeConverter<Board, String> {

    override fun convertToDatabaseColumn(board: Board): String {
        return try {
            OBJECT_MAPPER.writeValueAsString(board)
        } catch (e: JsonProcessingException) {
            throw RuntimeException(e)
        }
    }

    override fun convertToEntityAttribute(dbJson: String): Board {
        return try {
            OBJECT_MAPPER.readValue(dbJson, Board::class.java)
        } catch (e: JsonProcessingException) {
            throw RuntimeException(e)
        }
    }

    companion object {
        private val OBJECT_MAPPER = ObjectMapper().apply {
            setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
            setVisibility(PropertyAccessor.IS_GETTER, JsonAutoDetect.Visibility.NONE)
        }
    }
}
